cc3d/core/XMLUtils.py

```diff
@@ -3,25 +3,31 @@
 from cc3d.cpp.CC3DXML import *
 
 
-def dictionaryToMapStrStr(_dictionary:dict)->object:
+def dictionaryToMapStrStr(_dictionary: dict) -> object:
     """
-    converts python dictionaty to c++ map (std::map<std:string, std:string>)
+    converts python dictionary to c++ map (std::map<std:string, std:string>)
     :param _dictionary:
     :return:
     """
-    mapStrStr = CC3DXML.MapStrStr()
-    for key in _dictionary.keys():
-        # mapStrStr[key.encode()]=str(_dictionary[key])
-        mapStrStr[key] = str(_dictionary[key])
-    return mapStrStr
+    try:
+        map_str_str = CC3DXML.MapStrStr()
+        for key in _dictionary.keys():
+            map_str_str[key] = str(_dictionary[key])
+        return map_str_str
+    except TypeError:
+        return CC3DXML.MapStrStr({str(k): str(v) for k, v in _dictionary.items()})
 
 
 class ElementCC3D:
-    def __init__(self, _name, _attributes={}, _cdata=""):
+    def __init__(self, _name, _attributes: dict = None, _cdata=""):
+        if _attributes is None:
+            _attributes = {}
```

Comment:
Why use a default value of None?

Reply:
In general, it's good practice to avoid using mutable objects for default values. See [here](https://docs.python-guide.org/writing/gotchas/) for details.

Reply:
Nice find! I wasn't aware of that nuance.